docs/proposals/cluster-groups.md

```diff
@@ -0,0 +1,25 @@
+# Cluster Groups
+
+## Problem
+To allow cluster-scoped group definitions to be shared across multiple related projects, we'll introduce the concept of ClusterGroups (cluster-scoped sets of users) that are managed by cluster-admins.
+
+
+## Use Cases
+ 1.  I want to be able to define some ClusterGroups that will affect policy evaluation across all projects.
+ 1.  I want to be able to efficiently manage group membership.
+
+
+## ClusterGroups
+ClusterGroups are cluster-scoped sets of users that are managed by cluster-admins and may contain any user.  ClusterGroups can be bound to roles in all projects.  That means that if UserA is added to ClusterGroupB and ProjectC has a rolebinding that says "ClusterGroupB is an editor", UserA now has editor permissions in ProjectC.  ClusterGroups will be described using an API object that looks like this:
+```
+// ClusterGroup contains the set of users that belong to a given ClusterGroup.  The name may not contain ":"
+type ClusterGroup struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the ClusterGroup
+	Users util.StringSet
```

Comment:
Hrm - def should be consistent, although these can include users that
aren't objects.

On Jul 15, 2015, at 7:43 AM, David Eads <notifications@github.com> wrote:

In docs/proposals/cluster-groups.md
<https://github.com/openshift/origin/pull/3627#discussion_r34669781>:

> +
> +## Use Cases
> + 1.  I want to be able to define some ClusterGroups that will affect policy evaluation across all projects.
> + 1.  I want to be able to efficiently manage group membership.
> +
> +
> +## ClusterGroups
> +ClusterGroups are cluster-scoped sets of users that are managed by cluster-admins and may contain any user.  ClusterGroups can be bound to roles in all projects.  That means that if UserA is added to ClusterGroupB and ProjectC has a rolebinding that says "ClusterGroupB is an editor", UserA now has editor permissions in ProjectC.  ClusterGroups will be described using an API object that looks like this:
> +```
> +// ClusterGroup contains the set of users that belong to a given ClusterGroup.  The name may not contain ":"
> +type ClusterGroup struct {
> +	kapi.TypeMeta
> +	kapi.ObjectMeta
> +
> +	// Users is a list of Users that are members of the ClusterGroup
> +	Users util.StringSet

 []string for external

I had actually debated []LocalObjectReference for patchability, but most of
the code won't care, so I wasn't hung up on it.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3627/files#r34669781>.